toolflow/scala/src/main/scala/tapasco/activity/hls/VivadoHighLevelSynthesis.scala

```diff
@@ -70,10 +70,12 @@ private object VivadoHighLevelSynthesis extends HighLevelSynthesizer {
       }
 
       // execute Vivado HLS (max. runtime: 1 day)
-      val vivadoRet = InterruptibleProcess(Process(Seq("vivado_hls",
+      val vivado_hls_cmd = Seq("timeout", (cfg.hlsTimeOut getOrElse (24 * 60 * 60)).toString, "vivado_hls",
         "-f", script.toString,
-        "-l", logfile.toString
-      ), script.getParent.toFile), waitMillis = Some(24 * 60 * 60 * 1000))
+        "-l", logfile.toString)
+      val process = Process(vivado_hls_cmd, script.getParent.toFile) 
+      val vivadoRet = InterruptibleProcess(process,
+         waitMillis = Some(24 * 60 * 60 * 1000))
```

Comment:
How does the implementation avoid that `InterruptibleProcess` with the same time-limit as the `timeout` command kills the `timeout` command, before `timeout` has sent the appropriate signals to `vivado-hls`. Should the `InterruptibleProcess` maybe have a slightly longer time-limit (few seconds) to avoid killing the `timeout` command?

Besides that, it seems that the timeout of the `InterruptibleProcess` does not use `cfg.hlsTimeout` and would kill jobs if an HLS-timeout > 24h is specified.

Reply:
> maybe have a slightly longer time-limit (few seconds) to avoid killing the timeout command?

you are completely right! I thought I wrote it like so, but obviously I forgot to add that. Something like

```scala
waitMillis = Some( (cfg.hlsTimeOut getOrElse (24 * 60 * 60)) * 1000 + 1000)
```

should suffice. I will change that.